assets/app/view/tile.rb

```diff
@@ -47,15 +47,18 @@ def render
       children << render_tile_part(Part::Towns, routes: @routes) if @tile.towns.any?
 
       # OO tiles have different rules...
-      children << render_tile_part(Part::LocationName) if @tile.cities.size > 1 && @tile.location_name
+      children << render_tile_part(Part::LocationName) if @tile.location_name && @tile.cities.size > 1
+
+      # cities and towns render revenue, unless there are more than 2 cities on
+      # the tile, or the revenue varies by phase
+      if (@tile.cities.size > 2) || @tile.stops.any? { |s| s.revenue.values.uniq.size > 1 }
```

Comment:
looks like revenu.values.unique.size gets called a lot 

should we cache this in the revenue center?

Reply:
9488efb

Reply:
cached revenue values.uniq but didn't cache .size on it